### PR TITLE
nautilus: mgr/telemetry: Ignore crashes in report when module not enabled

### PR DESCRIPTION
**Pull request by wido**
The crash module is not guaranteerd to be enabled and this will render
the telemetry module useless:

  Error EINVAL: Traceback (most recent call last):
    File "/usr/lib/ceph/mgr/telemetry/module.py", line 325, in handle_command
      report = self.compile_report()
    File "/usr/lib/ceph/mgr/telemetry/module.py", line 291, in compile_report
      report['crashes'] = self.gather_crashinfo()
    File "/usr/lib/ceph/mgr/telemetry/module.py", line 214, in gather_crashinfo
      errno, crashids, err = self.remote('crash', 'do_ls', '', '')
    File "/usr/lib/ceph/mgr/mgr_module.py", line 845, in remote
      args, kwargs)
  ImportError: Module not found

We can safely ignore this error and just continue without the crash information.

Fixes: https://tracker.ceph.com/issues/42116

Signed-off-by: Wido den Hollander <wido@42on.com>

### CODE REVIEW

**Comment by wido** [comment]
@liewegas I see,  But somebody can still disable the module, right?

I actually found this issue with Mimic. So if this PR gets approved I was about to open a backport for Mimic.

**Comment by liewegas** [comment]
On Wed, 2 Oct 2019, Wido den Hollander wrote:
> @liewegas I see,  But somebody can still disable the module, right?

The always_on modules can't be disabled.  I suspect this is a mimic-only issue!

**Comment by wido** [comment]
Understood. I'll open a new PR for just Mimic which then hopefully makes it in to 13.2.7

**Comment by dmick** [comment]
....although it seems like this code increases the safety of the report anyway...

**Comment by wido** [comment]
@dmick It's very unlikely this happens in Nautilus. My idea was to cherry-pick this to Mimic once it merges for Nautilus.

**Comment by dmick** [comment]
I understand, but if we ever make the crash module not required, or a bug happens...it's just safety code, and not a performance problem, so...

**Comment by wido** [comment]
Can we then merge this one as it won't hurt? I'll then backport it to Mimic so that it makes it into 13.2.7

**Comment by smithfarm** [comment]
@wido - please set Milestone tag when opening PRs targeting stable branches

**Comment by smithfarm** [comment]
What happens if the crash module is explicitly disabled in master/octopus? The telemetry module will still load just fine?

**Comment by tserong** [comment]
> What happens if the crash module is explicitly disabled in master/octopus? The telemetry module will still load just fine?

Users can't explicitly disable the crash module, as it's hard coded to be always on. It'd take us removing it from the list of always on modules to disable it. If we did that to master/octopus, then we'd hit the same problem Wido saw with Mimic.

**Comment by tserong** [comment]
(Code looks fine BTW, no objection to merging it regardless of the list of always on modules)

**Comment by smithfarm** [comment]
> Users can't explicitly disable the crash module, as it's hard coded to be always on.

And this is the case with nautilus as well, right?

**Comment by smithfarm** [comment]
Before adding this "harmless safety code" (and I agree it does look harmless) directly to a stable branch, I would like to see an analysis of why the same code could not be added to master first and backported from there.

The same line that is being guarded with `try...except` here also exists in master [1] so, from my naive reading, the `try...except` guard could be applied to master without any risk, and with the same potential benefit as adding it directly to nautilus brings.

In other words: if the argument argument for adding it to nautilus is "it's harmless and might be useful under some unforeseen circumstances", then the same argument could be applied to master?

[1] https://github.com/ceph/ceph/blob/c2adfb62c62fb742529f6eb3bf9bccc13dd4efed/src/pybind/mgr/telemetry/module.py#L378

**Comment by tserong** [comment]
> And this is the case with nautilus as well, right?

Yep

**Comment by tserong** [comment]
> In other words: if the argument argument for adding it to nautilus is "it's harmless and might be useful under some unforeseen circumstances", then the same argument could be applied to master?

Yep. Actually, in master, the same applies to the `gather_device_report()` function, called a couple lines down, which relies on the devicehealth always on module.

**Comment by wido** [comment]
@smithfarm In master the Telemetry module has been large rewritten which does not suffer from this. It checks for enabled 'channels' for example.

So I can't fix this in master as the PR wouldn't merge back to Nautilus and Mimic.

**Comment by smithfarm** [comment]
@wido It doesn't need to cherry-pick without conflicts. The conflict resolution will be trivial.

**Comment by liewegas** [comment]
Yeah, just open the PR targeting mimic.
